pkg/cloudprovider/providers/gce/gce_addresses.go

```diff
@@ -85,3 +85,23 @@ func (gce *GCECloud) GetRegionAddress(name, region string) (*compute.Address, er
 	v, err := gce.service.Addresses.Get(gce.projectID, region, name).Do()
 	return v, mc.Observe(err)
 }
+
+// GetRegionAddressByIP returns the regional address matching the given IP
+// address. Note that if such an address does not exist, the function simply
+// returns a nil object. It does not return http.StatusNotFound like other
```

Comment:
I've also considered that. My only concern was that we record metrics for this calls. Faking errors would affect the metrics. If it's ok, I'll change the code.
(p.s., I also added this function to the the fake in this PR, even though we don't use it in unit test yet).